src/cephadm/cephadm.py

```diff
@@ -4692,6 +4696,34 @@ def command_list_images(ctx: CephadmContext) -> None:
     # print default images
     cp_obj.write(sys.stdout)
 
+
+def update_service_for_daemon(ctx: CephadmContext,
+                              available_daemons: list,
+                              update_daemon: list) -> None:
+    """ Update the unit.meta file of daemon with required service name for valid daemons"""
+
+    data = {'service_name': ctx.service_name}
+    # check if all the daemon names are invalid
+    if list(set(update_daemon) & set(available_daemons)) != update_daemon:
```

Comment:
sounds like you are trying to check if `update_daemons` is a subset of `update_daemon`, a direct way (and more readable) to do that is: 

`set(update_daemons).issubset(set(available_daemons))
`